packages/blade/src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,250 @@
+import { useChipGroupContext } from './ChipGroup/ChipGroupContext';
+import {
+  chipIconSizes,
+  chipTextSizes,
+  chipColorTokens,
+  getChipHoverTokens,
+  chipHeightTokens,
+  chipHorizontalPaddingTokens,
+} from './chipTokens';
+import { AnimatedChip } from './AnimatedChip';
+import type { IconComponent } from '~components/Icons';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { getStyledProps } from '~components/Box/styledProps';
+import type { StyledPropsBlade } from '~components/Box/styledProps';
+import BaseBox from '~components/Box/BaseBox';
+import { SelectorLabel } from '~components/Form/Selector/SelectorLabel';
+import { SelectorInput } from '~components/Form/Selector/SelectorInput';
+import type { BladeElementRef } from '~utils/useBladeInnerRef';
+import type { StringChildrenType, TestID } from '~utils/types';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { throwBladeError } from '~utils/logger';
+import { useCheckbox } from '~components/Checkbox/useCheckbox';
+import { useRadio } from '~components/Radio/useRadio';
+import { isReactNative, makeSize, useBreakpoint } from '~utils';
+import { Text } from '~components/Typography';
+import { useTheme } from '~components/BladeProvider';
+
+type OnChange = ({
+  isChecked,
+  event,
+  value,
+}: {
+  isChecked: boolean;
+  event?: React.ChangeEvent;
+  value?: string;
+}) => void;
+
+type ChipProps = {
+  /**
+   * Sets the label of the Chip
+   */
+  children: StringChildrenType;
+  /**
+   * Displays the Blade Icon component within the Chip
+   * Accepts a component of type `IconComponent` from Blade.
+   *
+   */
+  icon?: IconComponent;
+  /**
+   * Sets the Chip's visual intent
+   *
+   */
+  intent?: 'positive' | 'negative' | 'neutral';
+  /**
+   * If `true`, the Chip will be disabled
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * The value to be used in the Chip input.
+   * This is the value that will be returned on form submission.
+   * Use `onChange` to update its value
+   */
+  value?: string;
+} & TestID &
+  StyledPropsBlade;
+
+const _Chip: React.ForwardRefRenderFunction<BladeElementRef, ChipProps> = (
+  { isDisabled, value, children, icon: Icon, intent, testID, ...styledProps },
+  ref,
+) => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const groupProps = useChipGroupContext();
+  const isInsideGroup = !isEmpty(groupProps);
+  const [isPressed, setIsPressed] = React.useState(false);
+
+  if (__DEV__) {
+    if (!isInsideGroup) {
+      throwBladeError({
+        moduleName: 'Chip',
+        message:
+          '<Chip /> component should only be used within the context of a <ChipGroup /> component',
+      });
+    }
+  }
+
+  const _isDisabled = isDisabled ?? groupProps?.isDisabled;
+  const _name = groupProps?.name;
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```

Comment:
I think there is a bug in this rule because `value!` is actually converting the type from `string | undefined` to `string` here. I can see [multiple instances of this rule being disabled.](https://github.com/search?q=repo%3Arazorpay%2Fblade%20%40typescript-eslint%2Fno-unnecessary-type-assertion&type=code)